plugins/notifications-backend/src/service/router.ts

```diff
@@ -384,59 +426,68 @@ export async function createRouter(
           channel: 'notifications',
         });
       }
+      postProcessNotification(ret, opts);
     }
     return notifications;
   };
 
   // Add new notification
-  router.post('/', async (req, res) => {
-    const { recipients, payload } = req.body;
-    const notifications = [];
-    let users = [];
-
-    const credentials = await httpAuth.credentials(req, { allow: ['service'] });
+  router.post(
+    '/',
+    async (req: Request<any, Notification[], NotificationSendOptions>, res) => {
```

Comment:
Just a thought: By using blind casting the `request` to `Request<any, Notification[], ...` we can get false expectation that the received object is of the `Notification[]` type. IMHO, we should check the input and then eventually claim that it is of the expected type.

Reply:
Yeah, something to refactor I think for all endpoints. Or move to use OpenAPI Router.